keps/sig-windows/4802-windows-node-shutdown/README.md

```diff
@@ -0,0 +1,889 @@
+know that this has succeeded?
+-->
+*   Focus on handling shutdown on machines where kubelet itself runs as a Windows service, 
+    on machines where kubelet is loaded through other loaders such as nssm.exe, this feature 
+    will not be able to be enabled.
+*   Make kubelet aware of underlying node shutdown event and trigger pod
+    termination with sufficient grace period to shutdown properly
+*   Handle node shutdown in cloud-provider agnostic way
+*   Introduce minimal shutdown delay in order to shutdown node soon as possible
+    (but not sooner)
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+*   Let users modify or change existing pod lifecycle or introduce new inner
+    pod depencides / shutdown ordering
+*   Support kubelet itselft running as a non-Windows service (ex: loaded through nssm.exe), 
+    under those scenarios, kublet will run with this feature being disabled, while operators/developers
+    can get corresponding feature disabling information from the kublet logs
+*   Support Cancellation of Windows shutdown
+*   Provide guarantee to handle all cases of graceful node shutdown, for
+    example abrupt shutdown or sudden power cable pull can’t result in graceful
+    shutdown
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+*   As a cluster administrator, I can configure the nodes in my cluster to
+    allocate X seconds for my pods to terminate gracefully during a node
+    shutdown
+
+#### Story 2
+
+*   As a developer I can expect that my pods will terminate gracefully during
+    node shutdowns
+
+### Background on Windows Shutdown
+
+In the context of this KEP, shutdown is referred to as shutdown or restart of the
+underlying Windows machine. A shutdown can be initiated via a
+variety of methods for example:
+
+1. `shutdown /t 0`
+2. `shutdown /t 30` `#schedule a delayed shutdown in 30 seconds`
+3. `shutdown /r /t 30` `#schedule a delayed restart in 30 seconds`
+4. Physically pressing the power button on the machine
+5. If a machine is a VM, the underlying hypervisor can press the “virtual”
+   power button
+6. For a cloud instance, stopping the instance via Cloud API, e.g. via `az vm stop -g xx -n xx`.
+   Depending on the cloud provider, this may result in virtual power button press by the underlying hypervisor
```

Comment:
I suggest not using the "power button" analogy here but rather be generic and mention ACPI initiated shutdown / restart using the options provided by Windows.